reactor3/src/main/java/com/linecorp/armeria/common/reactor3/RequestContextHooks.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.reactor3;
+
+import java.util.function.Consumer;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscription;
+
+import com.linecorp.armeria.common.ContextHolder;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RequestContextStorage;
+import com.linecorp.armeria.common.util.SafeCloseable;
+
+import reactor.core.CoreSubscriber;
+import reactor.core.Disposable;
+import reactor.core.Scannable;
+import reactor.core.Scannable.Attr;
+import reactor.core.publisher.ConnectableFlux;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Hooks;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.ParallelFlux;
+import reactor.util.context.Context;
+
+/**
+ * Utility class to keep {@link RequestContext} during
+ * <a href="https://github.com/reactor/reactor-core">Reactor</a> operations.
+ */
+public final class RequestContextHooks {
+
+    private static final String ON_EACH_OPERATOR_HOOK_KEY =
+            RequestContextHooks.class.getName() + "#ON_EACH_OPERATOR_HOOK_KEY";
+
+    private static final String ON_LAST_OPERATOR_HOOK_KEY =
+            RequestContextHooks.class.getName() + "#ON_LAST_OPERATOR_HOOK_KEY";
+
+    private static boolean enabled;
+
+    /**
+     * Enables {@link RequestContext} during Reactor operations.
+     * The reactor {@link Publisher}s such as {@link Mono} and {@link Flux} will have the
+     * {@link RequestContext} which is in the {@link RequestContextStorage} when the {@link Publisher}s
+     * are created. Then, the {@link RequestContext} is propagated during the
+     * operations so that you can get the context using {@link RequestContext#current()}.
+     *
+     * <p>However, please note that {@link Mono#doOnCancel(Runnable)}, {@link Mono#doFinally(Consumer)},
+     * {@link Flux#doOnCancel(Runnable)} and {@link Flux#doFinally(Consumer)} will not propagate the context.
+     *
+     * <p>Also, note that this method does not have any relevance to {@link Context} API.
```

Comment:
How about `any relevance to Reactor's own` for clarity?

Reply:
Fixed. 😉